types/basetypes/float64_type.go

```diff
@@ -136,26 +136,33 @@ func (t Float64Type) ValueFromTerraform(ctx context.Context, in tftypes.Value) (
 
 	var bigF *big.Float
 	err := in.As(&bigF)
-
 	if err != nil {
 		return nil, err
 	}
 
-	f, accuracy := bigF.Float64()
+	// Copy the float retrieved from tftypes.Value to avoid any unexpected pass-by-reference shenanigans
+	copiedBigF := new(big.Float)
+	copiedBigF.Copy(bigF)
+
+	f, accuracy := copiedBigF.Float64()
```

Comment:
Turns out I was being overly cautious when describing this and tftypes already does the copying: https://github.com/hashicorp/terraform-plugin-go/blob/81f0d496aaa39e43cc185f5c961df5fcd86192da/tftypes/value.go#L449

Reply:
Good catch, removed copy in 4425e1c ⚡